wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -65,15 +65,27 @@ public void clearFirstRunDialogues () throws Exception {
         if (takeMeToBox.exists()) {
             clickUiObject(takeMeToBox, timeout);
         }
-        UiObject converationView = new UiObject(new UiSelector()
-                                            .resourceId("com.google.android.gm:id/conversation_list_view")
-                                            .className("android.widget.ListView"));
-        if (!converationView.waitForExists(networkTimeout)) {
-            throw new UiObjectNotFoundException("Could not find \"converationView\".");
+
+        UiObject syncNowButton = new UiObject(new UiSelector().textContains("Sync now")
+                                                              .className("android.widget.Button"));
+
+        // On some devices we need to wait for a sync to occur after clearing the data
+        // We also need to sleep here since waiting for a new window is not enough
+        if (syncNowButton.exists()) {
+            syncNowButton.clickAndWaitForNewWindow(timeout);
+            sleep(10);
```

Comment:
The sleep is a bit of a hack, but I couldn't find a nice work around. Let me know how this works when you test it on your devices.